### PR TITLE
Move snippets to async/await

### PR DESCRIPTION
**Pull request by morganchen12**
Includes some 4 space to 2 space indentation changes.

### CODE REVIEW

File: ml-functions/MLFunctionsExample/ViewController.m [hunk 1]
```diff
@@ -85,10 +85,10 @@ - (void)annotateImage:(NSDictionary *)requestData {
                             callWithObject:requestData
                                 completion:^(FIRHTTPSCallableResult * _Nullable result, NSError * _Nullable error) {
           if (error) {
-            if (error.domain == FIRFunctionsErrorDomain) {
+            if ([error.domain isEqualToString:@"com.firebase.functions"]) {
               FIRFunctionsErrorCode code = error.code;
               NSString *message = error.localizedDescription;
-              NSObject *details = error.userInfo[FIRFunctionsErrorDetailsKey];
+              NSObject *details = error.userInfo[@"details"];
```
Comment (morganchen12) [inline ml-functions/MLFunctionsExample/ViewController.m:91]: @paulb777 the functions constants don't seem to be importable into ObjC, so this file wouldn't build with the previous variable names.

File: firestore/swift/firestore-smoketest/SolutionAggregationViewController.swift [hunk 2]
```diff
@@ -21,83 +21,86 @@ import FirebaseFirestore
...
-        }
-
-    }
-
-    let arinell = Restaurant(name: "Arinell Pizza", avgRating: 4.65, numRatings: 683)
-    // [END restaurant_struct]
-
-    struct Rating {
-        let rating: Float
+    init(name: String, avgRating: Float, numRatings: Int) {
```
Comment (peterfriese) [inline firestore/swift/firestore-smoketest/SolutionAggregationViewController.swift:33]: Aren't these synthesised automatically?

File: firestore/swift/firestore-smoketest/SolutionArraysViewController.swift [hunk 3]
```diff
@@ -21,94 +21,94 @@ import FirebaseFirestore
...
+    super.viewDidLoad()
+    db = Firestore.firestore()
+  }
+
+  func queryInCategory() async {
+    // [START query_in_category]
+    do {
+      let snapshot = try await db.collection("posts")
+        .whereField("categories.cats", isEqualTo: true)
+        .getDocuments()
```
Comment (peterfriese) [inline firestore/swift/firestore-smoketest/SolutionArraysViewController.swift:36]: Let's add a line to indicate you might want to do something with the snapshot.

Also, in the completion handler sample, it's called `querySnapshot` - shouldn't we call it that, too?

File: firestore/swift/firestore-smoketest/SolutionArraysViewController.swift [hunk 4]
```diff
@@ -21,94 +21,94 @@ import FirebaseFirestore
...
+    // [END query_in_category_timestamp]
+  }
+
+  // [START post_with_array]
+  struct PostArray {
+
+    let title: String
+    let categories: [String]
+
+    init(title: String, categories: [String]) {
```
Comment (peterfriese) [inline firestore/swift/firestore-smoketest/SolutionArraysViewController.swift:63]: Same - aren't these synthesised by the compiler?

File: firestore/swift/firestore-smoketest/SolutionGeoPointViewController.swift [hunk 5]
```diff
@@ -23,83 +23,80 @@ import CoreLocation
...
-                // We have to filter out a few false positives due to GeoHash accuracy, but
-                // most will match
-                let distance = GFUtils.distance(from: centerPoint, to: coordinates)
-                if distance <= radiusInM {
-                    matchingDocs.append(document)
-                }
-            }
-        }
+    // After all callbacks have executed, matchingDocs contains the result. Note that this code
+    // executes all queries serially, which may not be optimal for performance.
```
Comment (peterfriese) [inline firestore/swift/firestore-smoketest/SolutionGeoPointViewController.swift:78]: Did you consider using a task group? 

https://www.hackingwithswift.com/quick-start/concurrency/how-to-create-a-task-group-and-add-tasks-to-it

**Review by peterfriese — APPROVED** [review]
LGTM with a couple of nits.

File: firestore/swift/firestore-smoketest/SolutionArraysViewController.swift (hunk 3 above)
Comment (morganchen12) [reply firestore/swift/firestore-smoketest/SolutionArraysViewController.swift:36]: Done.

File: firestore/swift/firestore-smoketest/SolutionArraysViewController.swift (hunk 4 above)
Comment (morganchen12) [reply firestore/swift/firestore-smoketest/SolutionArraysViewController.swift:63]: Done.

File: firestore/swift/firestore-smoketest/SolutionGeoPointViewController.swift (hunk 5 above)
Comment (morganchen12) [reply firestore/swift/firestore-smoketest/SolutionGeoPointViewController.swift:78]: Done.

**Pull request merged**